### ISSUE TITLE
[java] InvalidSlf4jMessageFormat does not handle inline incrementation of arguments

### ISSUE THREAD

**Issue by lidel** [pmd/pmd#365]
**Rule Set:**
Logging-Java/`InvalidSlf4jMessageFormat`

**Description:**

`InvalidSlf4jMessageFormat` produces false-positives when inline  `++value`  notation is used in arguments of `Logger.info` method.

This seems to be somehow related to https://github.com/pmd/pmd/issues/253, where value is also "calculated".

**Code Sample demonstrating the issue:**

```java
Logger LOG = LoggerFactory.getLogger(Test.class);
int attempt = 0;
LOG.info("test (attempt #{})", ++attempt);
```

produces bogus error:

```
Rule:InvalidSlf4jMessageFormat Priority:5 Missing arguments, expected 1 argument but have 0.
```

However this:

```java
Logger LOG = LoggerFactory.getLogger(Test.class);
int attempt = 0;
attempt += 1;
LOG.info("test (attempt #{})", attempt);
```

works fine.

**PMD Version:** *5.5.6*
**Running PMD through:** *Maven*

**Comment by jsotuyod** [comment]
@lidel thanks! This is definitely a bug!

Unlike #253, which is really hard (we need to figure out what the final string will look like), this one should be easy to fix.
